cli/src/semgrep/app/scans.py

```diff
@@ -42,6 +42,13 @@
 logger = getLogger(__name__)
 
 
+@dataclass
+class ScanCompleteResult:
+    success: bool
+    app_block_override: bool
```

Comment:
Also the type is not super precise ... Can we have app_block_override = false and app_block_reason different from an empty string?
Or reason has only a meaning when app_block_override is true? (which is what I assumed here: https://github.com/returntocorp/semgrep/blob/0d03423212f6d54146239a38fe7f7b8d23530e1c/src/osemgrep/cli_ci/Scan_helper.mli#L2 )

Reply:
I agree but there's so much about this that's not precise and that could be improved but that's out of scope of this change, we can follow up separately.